### PR TITLE
Authentication and authorisation

### PR DESCRIPTION
**Pull request by titopixelfusion**
- Integration of Laravel Passport for oauth

### CODE REVIEW

**Comment by jpcaparas** [comment]
I'll be updating this comment with the commands I ran to reproduce this feature:

1. Run `composer install`

File: app/Http/Controllers/Api/Oauth/CreateClientController.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?php
```
Comment (titopixelfusion) [inline app/Http/Controllers/Api/Oauth/CreateClientController.php]: Fixed, so it seems that if the in element is body, we must add a schema attribute. Not sure exactly what the value should be since in our case it is just json, not associated with any schema.

File: composer.json [hunk 2]
```diff
@@ -5,10 +5,11 @@
     "license": "MIT",
     "type": "project",
     "require": {
-        "php": ">=5.6.4 || >=7.0",
+        "php": ">=7.0",
```
Comment (titopixelfusion) [inline composer.json]: IMHO this should be part of another PR, we should also type hint return values, but that is out of scope.

File: database/factories/ModelFactory.php [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <?php
 
+use App\Models\User;
```
Comment (titopixelfusion) [inline database/factories/ModelFactory.php]: done. ordered imports in asc

File: storage/oauth-private.key [hunk 4]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (titopixelfusion) [inline storage/oauth-private.key]: I'd say there is no harm in just removing the files, they will still exist in the history, but since we are not using them live, we should be safe. do you agree @jpcaparas ?

File: app/Http/Controllers/Api/Oauth/CreateClientController.php [hunk 5]
```diff
@@ -47,20 +48,23 @@
...
  *     tags={"oauth"},
  *     summary="Create new password grant client",
  *     @SWG\Parameter(
+ * 			name="Authorization",
+ * 			in="header",
+ *          schema={"$ref": "#/definitions/authorization"},
+ * 			required=true,
+ * 			type="string",
+ * 			description="Authorization in header",
+ * 		),
```
Comment (ThijsBouwes) [inline app/Http/Controllers/Api/Oauth/CreateClientController.php]: For now i have added the authorization on every protected endpoint, is this the best way @jpcaparas @titopixelfusion

**Comment by jpcaparas** [comment]
@titopixelfusion for some reason it's throwing an exception when I run 

`php vendor/bin/codecept run functional OAuthCreateClientCept`

```php
#1  /Users/josephcaparas/Projects/cavalry-api/vendor/league/oauth2-server/src/AuthorizationServer.php:92
#2  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/passport/src/PassportServiceProvider.php:203
#3  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/passport/src/PassportServiceProvider.php:94
#4  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:716
#5  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:598
#6  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:567
#7  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Foundation/Application.php:702
#8  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:840
#9  /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:780
#10 /Users/josephcaparas/Projects/cavalry-api/vendor/laravel/framework/src/Illuminate/Container/Container.php:746
```

File: app/Http/Controllers/Api/Backup/Job/FetchController.php [hunk 6]
```diff
@@ -11,7 +11,7 @@
  *     tags={"BackupJob"},
  *     summary="Fetch an backup job",
  *     @SWG\Parameter(
- *          name="backupJob",
+ *          name="backupJobId",
```
Comment (titopixelfusion) [inline app/Http/Controllers/Api/Backup/Job/FetchController.php]: @jpcaparas  correct me if I'm wrong, but the parameter in the url is named backupJob
